src/main/scala/Tiebreaks.scala

```diff
@@ -0,0 +1,112 @@
+
+  def virtualOpponentScoreAtRound(p: Player, round: Int): Double =
+    scoreAtRound(p, round) + remainingRoundDraws(round)
+
+  def remainingRoundDraws(round: Int): Double =
+    ((tournament.nbRounds) - round - 1) * Draw
+
+  def score(g: Result): Double =
+    g match {
+      case Result(_, _, WonAgainst, _) => Win
+      case Result(_, _, DrewWith, _)   => Draw
+      case Result(_, _, LostTo, _)     => Loss
+
+      // Not sure any of the rest of these make any sense.
+      case _ => Loss
+    }
+
+  // When used for tie-breaking among players with the same raw score, no
+  // multiplying is necessary and the sum of the raw scores of the opponents
+  // played is used to break ties (Golombek 1977). When used as a tie-break
+  // system, it is equivalent to the Solkoff system.
+  // - https://en.wikipedia.org/wiki/Buchholz_system
+  def buccholz(p: Player): Double =
+    tournament.resultsForPlayer(p).map(buccholzForGame).sum
+
+  def buccholzForGame(g: Result): Double =
+    g match {
+      // Regular game played with regular result
+      case Result(_, Hero(Player(_), Present), _, Foe(p: Player)) => score(p)
+
+      case Result(round, Hero(p: Player, Present), WonAgainst, Foe(Virtual)) =>
+        virtualOpponentScoreAtRound(p, round)
+
+      // Not sure any of the rest of these make any sense.
+      case _                                                                 => 0
+    }
+
+  def sonnenbornBerger(p: Player): Double =
+    tournament.resultsForPlayer(p).map(sonnenbornBergerForGame).sum
+
+  def sonnenbornBergerForGame(g: Result): Double =
+    g match {
+      // Regular game played with regular result
+      case Result(_, Hero(Player(_), Present), WonAgainst, Foe(p: Player)) => score(p)
+      case Result(_, Hero(Player(_), Present), DrewWith, Foe(p: Player))   => score(p) / 2
+      case Result(_, Hero(Player(_), Present), LostTo, Foe(Player(_)))     => 0
+
+      // All other results are not worth anything for the current lila implementation
+      case Result(_, Hero(Player(_), Present), _, Foe(Virtual)) => 0
+
+      // Not sure any of the rest of these make any sense.
+      case _ => 0
+    }
+
+}
+
+object Tiebreak {
+  case class Round(r: Int) {
+    def bye(p: Player)                      = Result(r, Hero(p, Present), WonAgainst, Foe(Virtual))
+    def withdrawn(p: Player)                = Result(r, Hero(p, Absent), LostTo, Foe(Virtual))
+    def win(p: Player, o: Player)           = Result(r, Hero(p, Present), WonAgainst, Foe(o))
+    def lose(p: Player, o: Player)          = Result(r, Hero(p, Present), LostTo, Foe(o))
+    def draw(p: Player, o: Player)          = Result(r, Hero(p, Present), DrewWith, Foe(o))
+    def noShowDraw(p: Player)               = Result(r, Hero(p, Absent), DrewWith, Foe(Virtual))
+    def noShowLoss(p: Player, o: Player)    = Result(r, Hero(p, Absent), LostTo, Foe(o))
+    def absentLoss(p: Player)               = Result(r, Hero(p, Absent), LostTo, Foe(Virtual))
```

Comment:
Is this different to withdraw, as you could come back the round after - or does it drop you from the tournament?